### PR TITLE
Adding AddClientRoleComposite and DeleteClientRoleComposite methods

### PR DESCRIPTION
**Pull request by agorman**
These methods will allow for managing composite roles for clients.

It appears that these methods will work for both client and realm composite roles which the current `AddRealmRoleComposite` and `DeleteRealmRoleComposite` only work for realm composite roles.

### CODE REVIEW

File: client_test.go [hunk 1]
```diff
@@ -2343,6 +2343,56 @@ func TestGocloak_AddClientRoleToUser_DeleteClientRoleFromUser(t *testing.T) {
...
+	token := GetAdminToken(t, client)
+
+	tearDown, compositeRole := CreateClientRole(t, client)
+	defer tearDown()
+
+	tearDown, role := CreateClientRole(t, client)
+	defer tearDown()
+
+	compositeRoleModel, err := client.GetClientRole(token.AccessToken, cfg.GoCloak.Realm, gocloakClientID, compositeRole)
+	FailIfErr(t, err, "Can't get just created role with GetClientRole")
```
Comment (SVilgelm) [inline client_test.go]: please use `asserts.NoError` instead of `FailIfError`

**Comment by SVilgelm** [comment]
I think you can use only `TestGocloak_DeleteClientRoleComposite` to test both methods

File: client_test.go (hunk 1 above)
Comment (agorman) [reply client_test.go]: I'll get this on the first try one of these days.

Comment (SVilgelm) [reply client_test.go]: :)

**Comment by Nerzal** [comment]
LGTM pushed Tag v4.4.0

**Pull request merged**